### ISSUE TITLE
Deciphering the ideal input tensor size for a Monai defined network

### ISSUE THREAD

**Issue by saruarlive** [Project-MONAI/MONAI#1118]
Dea Nic-Ma et al,

Could you kindly explain what is the ideal input tensor size in a DataLoader to feed into a Monai defined network (eg. monai.networks.nets.UNet)?  

Should it be **_batch,channel,height,width,depth_**  or  **_batch,channel,depth,height,width_**? 

 Is it the same with PyTorch as PyTorch expects **_batch,channel,depth,height,width_** in their convolution and its related functions (eg, nn.Conv3d)?

Looking forward to hearing from you.

**Comment by Nic-Ma** [comment]
Hi @saruarlive ,

Thanks for your interest here.
We just follow the PyTorch style, it should be **_batch, channel, height, width, depth_** for the networks.

Thanks.

**Comment by saruarlive** [comment]
Dea Nic-Ma et al,

Thank you for your quick reply. you have said the tensor should be _**batch, channel, height, width, depth**_ for the networks. 
But in the Pytorch documentation, it is written like _**batch, channel, depth, height, width**_ as shown below.

![image](https://user-images.githubusercontent.com/24542592/96424037-70638a80-11fa-11eb-9667-9d07fff72ccb.png)

Could you kindly explain a bit more what norms of dimension should we follow?

**Comment by Nic-Ma** [comment]
Hi @saruarlive ,

I think PyTorch CONV3D is for general 3D data, so it depends on which dim you defined as `D` in the medical image.
@ericspod could you please help share more information about this question? I think you are expert about it.

Thanks.

**Comment by saruarlive** [comment]
Hi, @ericspod and @Nic-Ma,

Thanks for your quick response.
I am defining _**d**_ as slices of a nifti (MRI) image.For example, a tensor like (height=256, width=256, slices=155)
In this regard, what norms should I follow?

**Comment by Nic-Ma** [comment]
Hi @saruarlive ,

For your case, I think it matches MONAI dims(HWD) where D is the slices.
Thanks.

**Comment by saruarlive** [comment]
Hi @Nic-Ma 
Thanks for your reply,
Do you think the result will affect if I use (DHW) norms where D is the slice?
Looking forward to getting your intuition?

**Comment by ericspod** [comment]
I don't think it'll matter if you have data in whatever dimension ordering. If you have anisotropic data where the depth dimension is sparser than the other two a network will learn this but it wouldn't matter which dimension this was. Whether DHW or HWD makes sense relative to the original data depends on the format, Nifti for example actually stores as HWD so the depth dimension being last is what we'd expect there. We might want to standardize our descriptions in MONAI however to match Pytorch's ordering.

**Comment by wyli** [comment]
for the record pytorch has an inconsistent convention in the `grid_sample` https://discuss.pytorch.org/t/surprising-convention-for-grid-sample-coordinates/79997
